zipline/testing/fixtures.py

```diff
@@ -530,21 +531,25 @@ def init_class_fixtures(cls):
         super(WithTradingCalendars, cls).init_class_fixtures()
 
         cls.trading_calendars = {}
-
-        for cal_str in (
-            set(cls.TRADING_CALENDAR_STRS) |
-            {cls.TRADING_CALENDAR_PRIMARY_CAL}
-        ):
-            # Set name to allow aliasing.
-            calendar = get_calendar(cal_str)
-            setattr(cls,
-                    '{0}_calendar'.format(cal_str.lower()), calendar)
-            cls.trading_calendars[cal_str] = calendar
-
-        type_to_cal = iteritems(cls.TRADING_CALENDAR_FOR_ASSET_TYPE)
-        for asset_type, cal_str in type_to_cal:
-            calendar = get_calendar(cal_str)
-            cls.trading_calendars[asset_type] = calendar
+        # Silence `pandas.errors.PerformanceWarning: Non-vectorized DateOffset
+        # being applied to Series or DatetimeIndex` in trading calendar
+        # construction. This causes nosetest to fail.
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", PerformanceWarning)
```

Comment:
Could we wrap just the `get_calendar` calls, or is more needed? I think this only matters for subclasses of [`WithWerror`](https://github.com/quantopian/zipline/blob/860a065aede4602757b15a47eba71719afa48cec/zipline/testing/fixtures.py#L2071) (there are only 2). Think it's worth keeping that mixin @llllllllll ?